java/code/src/com/redhat/rhn/frontend/xmlrpc/channel/software/ChannelSoftwareHandler.java

```diff
@@ -2422,20 +2422,24 @@ public int regenerateNeededCache(User loggedInUser) {
      * Regenerate the yum cache for a specific channel.
      * @param loggedInUser The current user
      * @param channelLabel the channel label
+     * @param forceStr force regeneration
      * @return int - 1 on success!
      *
      * @xmlrpc.doc Regenerate yum cache for the specified channel.
      * @xmlrpc.param #session_key()
      * @xmlrpc.param #param_desc("string", "channelLabel", "the label of the
      *          channel")
+     * @xmlrpc.param #param_desc("string", "forceStr", "Y/N force cache regeneration")
      * @xmlrpc.returntype  #return_int_success()
      *
      */
-    public int regenerateYumCache(User loggedInUser, String channelLabel) {
+    public int regenerateYumCache(User loggedInUser, String channelLabel, String forceStr) {
```

Comment:
Is there a particular reason why a boolean is not used here?

Reply:
Good point, changed to boolean.